src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -98,192 +89,270 @@ private InventoryUtils()
     }
 
     /**
+     * Compares whether or not the item in an itemstack is equal to a given
+     * item.
+     *
+     * @param itemStack  ItemStack to check.
+     * @param targetItem Item to check.
+     * @param itemDamage the item damage value.
+     * @return True when item in item stack is equal to target item.
+     */
+    private static boolean compareItems(@Nullable final ItemStack itemStack, final Item targetItem, int itemDamage)
+    {
+        return !isItemStackEmpty(itemStack) && itemStack.getItem() == targetItem && (itemStack.getItemDamage() == itemDamage || itemDamage == -1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'itemStack' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)